#168284317 Refractor code for modularity

BurnerB opened the pull request:
#### What does this PR do?
* Increases app modularity and keeps it DRY

#### Description of Task to be completed
-create modular classes
-make  some functions middleware
-Remove hashed statements
-Add placeholderdata

#### How should this be manually tested?
* Click of the maintainability badge on the readme

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
[#168284317](https://www.pivotaltracker.com/story/show/168284317)

#### Screenshots (if appropriate)

BurnerB committed:
ch(modular):refractor code
-create modular classes
-make  some functions middleware
-Remove hashed statements
-Add placeholderdata
[Finish #168284317]

BurnerB committed:
ch(modularity):fix low coverage
-modify auth tests
-modify mentor tests
[finish #68284317]

Oluwasegun-AA commented on server/tests/auth.test.js:
```diff
@@ -43,6 +43,28 @@ describe('AUTH', () => {
+    it('should unsuccessfully sign up a user twice', (done) => {
+      chai.request(app)
+        .post('/api/v1/auth/signup')
+        .send({
```
@BurnerB , please create a separate folder in your `test` folder and name it `mocks` inside itn create a file called i.e `mocks.js`. inside it you will create instances of data to be used in your test file. It is not ethical to write the data in the test file. rather import the data from another file. please use this in al scenarios including for tokens

BurnerB replied:
ok,i'll work on this

BurnerB committed:
ch(modularity):add mock data
-add user mock data
-add session mock data
-add jwt mock data
[Finish #168284317]

BurnerB committed:
ch(modularity)add review route
-add review route
-utilize modular baseclass
[finishes #168284317]

BurnerB committed:
ft(modularity)fix failing build[Finishes #168284317]

BurnerB committed:
ch(modularity):fix ismentor bug[# #168284317]

BurnerB committed:
ch(modularity):review bug
[finishes #168284317]
